build.gradle

```diff
@@ -1,16 +1,93 @@
 plugins {
     id 'java'
+    id 'checkstyle'
+    id 'pmd'
+    id 'jacoco'
+    id 'com.github.johnrengelman.shadow' version '5.2.0'
 }
 
+group 'no.unit.nva'
+version '1.0-SNAPSHOT'
+
+sourceCompatibility = 1.8  // source-code version and must be <= targetCompatibility
+targetCompatibility = 1.8 // bytecode target version
+
 repositories {
     mavenCentral()
 }
 
 dependencies {
-    compile group: 'com.google.code.gson', name: 'gson', version: '2.8.6'
-    compile group: 'org.slf4j', name: 'slf4j-api', version:'1.7.29'
-    compile group: 'org.jsoup', name: 'jsoup', version:'1.12.1'
-    testCompile group: 'ch.qos.logback', name: 'logback-classic', version:'1.2.3'
+    implementation group: 'com.google.code.gson', name: 'gson', version: '2.8.6'
+    implementation group: 'org.slf4j', name: 'slf4j-api', version:'1.7.29'
+    implementation group: 'org.jsoup', name: 'jsoup', version:'1.12.1'
     implementation 'com.amazonaws:aws-lambda-java-core:1.2.0'
     testImplementation 'junit:junit:4.12'
+
+    shadowJar { zip64 true }
+}
+
+
+test {
+    failFast = true
+    finalizedBy jacocoTestReport
+}
+
+// We don't want a jar, just a fatJar
+jar.enabled = false
+project.tasks.build.dependsOn project.tasks.shadowJar
+
+pmd {
+    toolVersion = '6.17.0'
+    ruleSetConfig = rootProject.resources.text.fromFile('config/pmd/ruleset.xml')
+    ruleSets = []
+    ignoreFailures = false
+}
+
+checkstyle {
+    configFile = rootProject.resources.text.fromFile('config/checkstyle/checkstyle.xml').asFile()
+    showViolations = true
+    toolVersion '8.21'
+}
+
+tasks.withType(Checkstyle) {
+    reports {
+        xml.enabled false
+        html.enabled true
+        html.stylesheet rootProject.resources.text.fromFile('config/checkstyle/checkstyle-simple.xsl')
+    }
+}
+
+jacocoTestReport {
+    reports {
+        xml.enabled false
+        csv.enabled false
+        html.destination file("${buildDir}/reports/jacoco")
+    }
+}
+
+jacoco {
+    toolVersion = "0.8.4"
+}
+
+check.dependsOn jacocoTestCoverageVerification
+jacocoTestCoverageVerification.dependsOn(jacocoTestReport)
+
+jacocoTestCoverageVerification {
+    violationRules {
+        rule {
+            limit {
+                counter = 'METHOD'
+                value = 'COVEREDRATIO'
+                minimum = 0.75
```

Comment:
Needs to be 1

Reply:
I could not do better than 0.91

Reply:
fixed in next refactoring - pull request "refactoring..."